pkg/restapi/v1/oidc4vc/controller.go

```diff
@@ -125,12 +150,22 @@ func (c *Controller) GetOidcAuthorize(e echo.Context, params GetOidcAuthorizePar
 	req := e.Request()
 	ctx := req.Context()
 
+	if params.OpState == nil || len(*params.OpState) == 0 {
+		return apiUtil.WriteOutput(e)(nil, errors.New("op_state is required"))
+	}
+
 	ar, err := c.oauth2Provider.NewAuthorizeRequest(ctx, req)
 	if err != nil {
 		return resterr.NewFositeError(resterr.FositeAuthorizeError, e, c.oauth2Provider, err).WithAuthorizeRequester(ar)
 	}
 
-	// TODO: login and get consent from the user
+	mapped, mapOk := ar.(*fosite.AuthorizeRequest)
```

Comment:
is it possible that ar will not be of type `*fosite.AuthorizeRequest`, can this be replaced with `ar.(*fosite.AuthorizeRequest).State = *params.OpState`?

Reply:
I think no, but it's an external library, so we can not be 100% sure. it safer to properly handle that, rather than have panic